index.ts

```diff
@@ -1,24 +1,58 @@
 
-// Вам потрібно створити умовний тип, що служить для встановлення типу, що повертається з функції. 
-// Як параметр типу повинен обов'язково виступати функціональний тип.
 
-type ReturnFunctionType<T> = T extends () => infer returnType ? returnType : undefined;
+//Вам потрібно написати функцію, яка повертатиме об'єкт, де буде властивість result і це буде паліндром, 
+//і властивість steps — це число викликів до знаходження паліндрома. Для того, щоб перевірити себе 
+//використовуйте число 196. Це так зване Lychrel number — число яке немає поліндрому.
 
-function testFunction(): string {
-  return 'bla';
+function isPalindrome(num: number): boolean {
+  let strNum = num.toString();
+  return strNum === strNum.split('').reverse().join('');
 }
 
-type TestReturnType = ReturnFunctionType<typeof testFunction>;
-
+function generatePalindrome(num: number, maxSteps: number): { result: number | null; steps: number } {
```

Comment:
I just wanted to share an idea with you. Don't you think it would be great if we implement the parameter "maxSteps" with a default value? This would make things much easier and more efficient for everyone.